### PR TITLE
Enable different muxers

### PR DESCRIPTION
**Pull request by diasdavid**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -21,9 +22,16 @@ class Node extends libp2p {
...
+        muxer: process.env.LIBP2P_MUXER ? (() => {
+          const muxerPrefs = process.env.LIBP2P_MUXER
+          return muxerPrefs.split(',').map((pref) => {
+            switch (pref) {
+              case 'spdy': return spdy
+              case 'multiplex': return multiplex
+              default: throw new Error(pref + ' muxer not available')
+            }
+          })
+        })() : [spdy],
```
Comment (diasdavid) [inline src/index.js]: @VictorBjelkholm this is what I had in mind to enable both muxers (check also package.json)

Comment (dignifiedquire) [inline src/index.js]: How would this be configured in the browser and from js-ipfs?

Comment (diasdavid) [inline src/index.js]: Right now it means `Default to spdy`, that might change if multiplex becomes the 'muxer', until them, the env variables help for running tests.

Comment (VictorBjelkholm) [inline src/index.js]: @dignifiedquire for browser tests we can inject environment variables, with something like https://github.com/webpack/docs/wiki/list-of-plugins#defineplugin for webpack at least.

File: examples/echo/dialer.js [hunk 2]
```diff
@@ -1,65 +1,51 @@
...
 
-  const peerListener = new PeerInfo(ids[1])
-  idListener = ids[1]
-  peerListener.multiaddr.add(multiaddr('/ip4/127.0.0.1/tcp/10333'))
-  nodeDialer.start((err) => {
-    if (err) {
-      throw err
-    }
+  // Dialer
+  const idD = ids[0]
```
Comment (VictorBjelkholm) [inline examples/echo/dialer.js]: Could we keep the longer identifier? `dialerID` is a lot easier to read than `idD`

Comment (diasdavid) [inline examples/echo/dialer.js]: sure :)

**Pull request merged**